core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -304,12 +304,16 @@ private void reportMetrics(
       MetricsReport report,
       Supplier<Map<String, String>> headers) {
     reporter.report(report);
```

Comment:
we don't include this line into the `try-catch` because it's just using the `LoggingMetricsReporter`. Every metrics reporter should have its own "don't-fail-scan-planning-if-reporting-fails" and below is the one we're adding for REST metric reporting